src/icrar/leap-accelerate/algorithm/cpu/PhaseRotate.cc

```diff
@@ -130,16 +137,36 @@ namespace cpu
             icrar::cpu::RotateVisibilities(integration, metadata);
             output_integrations.emplace_back(direction, integration.integration_number, boost::none);
         }
-        auto avg_data_angles = metadata.avg_data.unaryExpr([](std::complex<double> c) -> Radians { return std::arg(c); });
-        auto& indexes = metadata.GetI1();
-        auto avg_data_t = avg_data_angles(indexes, 0); // 1st pol only
 
-        auto cal1 = metadata.GetAd1() * avg_data_t;
+        auto avg_data_angles = metadata.avg_data.unaryExpr([](std::complex<double> c) -> Radians { return std::arg(c); });    
+        Eigen::VectorXi indices1 = metadata.GetI1();
+        for(int& v : indices1)
+        {
+            if(v < 0)
+            {
+                v += avg_data_angles.rows(); /// -behaviour in python is to wrap around
```

Comment:
The last value in I1 is always -1 and this is updated to the LEAP-Cal behaviour, however I suspect the behaviour here may have been unintended. This only affects the calibration by ~10% buyt I'll chat with Richard and raise a ticket for changing to the intentional behaviour.